pkg/kubefedctl/orphaning/orphaning.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package orphaning
+
+import (
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
+
+	ctlutil "sigs.k8s.io/kubefed/pkg/controller/util"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/enable"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/options"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/util"
+)
+
+type orphanResource struct {
+	options.GlobalSubcommandOptions
+	typeName          string
+	resourceName      string
+	resourceNamespace string
+}
+
+// Bind adds the join specific arguments to the flagset passed in as an argument.
+func (o *orphanResource) Bind(flags *pflag.FlagSet) error {
+	flags.StringVarP(&o.resourceNamespace, "namespace", "n", "", "If present, the namespace scope for this CLI request")
+	err := flags.MarkHidden("kubefed-namespace")
+	if err != nil {
+		return err
+	}
+	err = flags.MarkHidden("dry-run")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewCmdOrphaning the head of orphaning-deletion sub commands
+func NewCmdOrphaning(cmdOut io.Writer, config util.FedConfig) *cobra.Command {
+	//	opts := &orphanResource{}
+	cmd := &cobra.Command{
+		Use:   "orphaning-deletion",
+		Short: "Manage orphaning delete policy",
+		Long:  "Manage orphaning delete policy",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cmd.Help()
+			if err != nil {
+				klog.Fatalf("Error: %v", err)
+			}
+		},
+	}
+	cmd.AddCommand(newCmdEnableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdDisableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdStatusOrphaning(cmdOut, config))
+
+	return cmd
+}
+
+//  Complete ensures that options are valid and marshals them if necessary.
+func (o *orphanResource) Complete(args []string, config util.FedConfig) error {
+
+	if len(args) == 0 {
+		return errors.New("resource type is required")
+	}
+
+	o.typeName = args[0]
+
+	if len(args) == 1 {
+		return errors.New("resource name is required")
+	}
+	o.resourceName = args[1]
+
+	return o.SetNamespace(config)
+}
+
+// If namespace was not set as an argument, it will be the current context namespace
+func (o *orphanResource) SetNamespace(config util.FedConfig) error {
+
+	if o.resourceNamespace != "" {
+		return nil
+	}
+	ns, _, err := config.GetClientConfig(o.HostClusterContext, o.Kubeconfig).Namespace()
```

Comment:
I think it would be preferable to follow the convention of other kubefedctl commands (e.g. federate) which default the resource namespace to `default` if it is not provided. The approach here might be nicer, but I think consistency is preferable and common support for discovering the namespace from the context can be implemented in a subsequent PR.

Reply:
done, will postpone namespace discovery for a separate PR, which is consistent with the `kubectl` behavior.